### DISCUSSION TITLE
How to execute python script in CORE emulator ?

### DISCUSSION THREAD

**Discussion by meenakshi510** [coreemu/core#613] [Q&A] [answered]
I just tried to export and execute the python script in the core emulator but it fails. Suggest, how to run python script on all nodes simultaneously in wired networks.

**Comment by PatrickTague** [comment]
i think the best way to do anything on all core nodes simultaneously would be to create a custom service.  there is some documentation available in this git about how to do that.  once you have a custom service, you can either enable it manually or via .xml per node, or you can enable it on all nodes of a particular type.  the documentation about node types is also helpful if you want to do that.

**Comment by meenakshi510** [comment]
Dear sir

As per your suggestion, how I can make changes in custom services and which
type of customized services should be used to run the script in all nodes?
With Regards
Meenakshi

**Comment by PatrickTague** [comment, marked as answer]
i would recommend reading this: https://coreemu.github.io/core/services.html#new-services

you an use the `UserDefined` service to run a shell script to do whatever you want, or you can create your own custom services (explained with examples on the same page linked above).  several of the sample networks in the .imn files packaged with core include examples you can look at.  for example, the `sample5-mgen.imn` network uses a `UserDefined` service to call the `mgen` traffic generation executable from a bash script.

**Reply by meenakshi510** [reply]
Ok Sir, I Will try this.
